core/src/test/java/forklift/consumer/injection/ConsumerTest.java

```diff
@@ -127,6 +134,63 @@ public void injectJson() {
         assertEquals("default", ec.strval);
     }
 
+    /**
+     * Tests constructor injection into a consumer.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void jsonConstructorInjection() throws Exception {
+        ConsumerService service = new ConsumerService(ServiceBeanResolver.class);
+        Consumer test = new Consumer(ConstructorJsonConsumer.class, forklift, this.getClass().getClassLoader());
+        test.setServices(Arrays.asList(service));
```

Comment:
Minor: Also, maybe it'd be nice to test service-based injection separately from forklift-based injection